[Native] Compilation fails with quarkus-google-cloud-firebase-admin

### Describe the Issue

I'm building a project using the quarkus-google-cloud-services extension pack and while it works perfectly in JVM mode I cannot compile my application to native due to failures.

### GraalVM Version

java version "23.0.2" 2025-01-21
Running Quarkus native-image plugin on GRAALVM 24.1 JDK 23.0.1+11-jvmci-b01

### Operating System and Version

MacOS 15.3.1

### Troubleshooting Confirmation

Followed the documentation.

I've also tried various combinations of adding the --initialize-at-run-time=org.apache.commons.logging.impl.Log4JLogger parameter without any joy.

Would like confirmation its not just me :)

### Expected Behavior

Application should compile to a native executable

### Actual Behavior

Native compilation fails with:

```

Error: com.oracle.svm.core.util.UserError$UserException: Class initialization of org.apache.commons.logging.impl.Log4JLogger failed. This error is reported at image build time because class org.apache.commons.logging.impl.Log4JLogger is registered for linking at image build time by command line and command line. Use the option 

    '--initialize-at-run-time=org.apache.commons.logging.impl.Log4JLogger'

 to explicitly request initialization of this class at run time.
Caused by: com.oracle.svm.core.util.UserError$UserException: Class initialization of org.apache.commons.logging.impl.Log4JLogger failed. This error is reported at image build time because class org.apache.commons.logging.impl.Log4JLogger is registered for linking at image build time by command line and command line. Use the option 

    '--initialize-at-run-time=org.apache.commons.logging.impl.Log4JLogger'

 to explicitly request initialization of this class at run time.
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:97)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.ensureClassInitialized(ClassInitializationSupport.java:204)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.computeInitKindAndMaybeInitializeClass(ClassInitializationSupport.java:428)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.computeInitKindAndMaybeInitializeClass(ClassInitializationSupport.java:366)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.maybeInitializeAtBuildTime(ClassInitializationSupport.java:178)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.maybeInitializeAtBuildTime(ClassInitializationSupport.java:167)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.SVMHost.onTypeReachable(SVMHost.java:350)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisUniverse.onTypeReachable(AnalysisUniverse.java:676)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.lambda$new$0(AnalysisType.java:336)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AnalysisFuture.ensureDone(AnalysisFuture.java:69)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.ensureOnTypeReachableTaskDone(AnalysisType.java:750)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.onReachable(AnalysisType.java:629)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AtomicUtils.atomicSetAndRun(AtomicUtils.java:49)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.registerAsReachable(AnalysisType.java:591)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.registerClass(ReflectionDataBuilder.java:256)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.lambda$register$6(ReflectionDataBuilder.java:202)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.lambda$runConditionalInAnalysisTask$2(ReflectionDataBuilder.java:184)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:166)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:152)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1726)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1717)
        at java.base/java.util.concurrent.ForkJoinTask$InterruptibleTask.exec(ForkJoinTask.java:1641)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1458)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2034)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:189)
Caused by: java.lang.NoClassDefFoundError: org/apache/log4j/Priority
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1161)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.ensureClassInitialized(ClassInitializationSupport.java:194)
        ... 26 more
Caused by: java.lang.ClassNotFoundException: org.apache.log4j.Priority
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:528)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageClassLoader.loadClass(NativeImageClassLoader.java:637)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:528)
        ... 29 more
Caused by: java.lang.NoClassDefFoundError: org/apache/log4j/Priority
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1161)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.ensureClassInitialized(ClassInitializationSupport.java:194)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.computeInitKindAndMaybeInitializeClass(ClassInitializationSupport.java:428)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.computeInitKindAndMaybeInitializeClass(ClassInitializationSupport.java:366)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.maybeInitializeAtBuildTime(ClassInitializationSupport.java:178)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.maybeInitializeAtBuildTime(ClassInitializationSupport.java:167)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.SVMHost.onTypeReachable(SVMHost.java:350)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisUniverse.onTypeReachable(AnalysisUniverse.java:676)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.lambda$new$0(AnalysisType.java:336)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AnalysisFuture.ensureDone(AnalysisFuture.java:69)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.ensureOnTypeReachableTaskDone(AnalysisType.java:750)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.onReachable(AnalysisType.java:629)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AtomicUtils.atomicSetAndRun(AtomicUtils.java:49)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.registerAsReachable(AnalysisType.java:591)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.registerClass(ReflectionDataBuilder.java:256)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.lambda$register$6(ReflectionDataBuilder.java:202)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.lambda$runConditionalInAnalysisTask$2(ReflectionDataBuilder.java:184)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:166)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:152)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1726)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1717)
        at java.base/java.util.concurrent.ForkJoinTask$InterruptibleTask.exec(ForkJoinTask.java:1641)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1458)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2034)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:189)
Caused by: java.lang.ClassNotFoundException: org.apache.log4j.Priority
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:528)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageClassLoader.loadClass(NativeImageClassLoader.java:637)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:528)
        ... 29 more
Caused by: java.lang.ClassNotFoundException: org.apache.log4j.Priority
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:528)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageClassLoader.loadClass(NativeImageClassLoader.java:637)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:528)
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1161)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.ensureClassInitialized(ClassInitializationSupport.java:194)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.computeInitKindAndMaybeInitializeClass(ClassInitializationSupport.java:428)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.computeInitKindAndMaybeInitializeClass(ClassInitializationSupport.java:366)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.maybeInitializeAtBuildTime(ClassInitializationSupport.java:178)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.maybeInitializeAtBuildTime(ClassInitializationSupport.java:167)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.SVMHost.onTypeReachable(SVMHost.java:350)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisUniverse.onTypeReachable(AnalysisUniverse.java:676)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.lambda$new$0(AnalysisType.java:336)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AnalysisFuture.ensureDone(AnalysisFuture.java:69)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.ensureOnTypeReachableTaskDone(AnalysisType.java:750)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.onReachable(AnalysisType.java:629)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AtomicUtils.atomicSetAndRun(AtomicUtils.java:49)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.registerAsReachable(AnalysisType.java:591)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.registerClass(ReflectionDataBuilder.java:256)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.lambda$register$6(ReflectionDataBuilder.java:202)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.lambda$runConditionalInAnalysisTask$2(ReflectionDataBuilder.java:184)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:166)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:152)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1726)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1717)
        at java.base/java.util.concurrent.ForkJoinTask$InterruptibleTask.exec(ForkJoinTask.java:1641)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1458)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2034)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:189)
Internal exception: com.oracle.graal.pointsto.util.AnalysisError: com.oracle.svm.core.util.UserError$UserException: Class initialization of org.apache.commons.logging.impl.Log4JLogger failed. This error is reported at image build time because class org.apache.commons.logging.impl.Log4JLogger is registered for linking at image build time by command line and command line. Use the option 

    '--initialize-at-run-time=org.apache.commons.logging.impl.Log4JLogger'

 to explicitly request initialization of this class at run time.
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AnalysisFuture.setException(AnalysisFuture.java:55)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:322)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AnalysisFuture.ensureDone(AnalysisFuture.java:69)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.ensureOnTypeReachableTaskDone(AnalysisType.java:750)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.onReachable(AnalysisType.java:629)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AtomicUtils.atomicSetAndRun(AtomicUtils.java:49)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.registerAsReachable(AnalysisType.java:591)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.registerClass(ReflectionDataBuilder.java:256)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.lambda$register$6(ReflectionDataBuilder.java:202)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.reflect.ReflectionDataBuilder.lambda$runConditionalInAnalysisTask$2(ReflectionDataBuilder.java:184)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:166)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:152)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1726)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1717)
        at java.base/java.util.concurrent.ForkJoinTask$InterruptibleTask.exec(ForkJoinTask.java:1641)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1458)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2034)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:189)
Caused by: com.oracle.svm.core.util.UserError$UserException: Class initialization of org.apache.commons.logging.impl.Log4JLogger failed. This error is reported at image build time because class org.apache.commons.logging.impl.Log4JLogger is registered for linking at image build time by command line and command line. Use the option 

    '--initialize-at-run-time=org.apache.commons.logging.impl.Log4JLogger'

 to explicitly request initialization of this class at run time.
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:97)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.ensureClassInitialized(ClassInitializationSupport.java:204)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.computeInitKindAndMaybeInitializeClass(ClassInitializationSupport.java:428)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.computeInitKindAndMaybeInitializeClass(ClassInitializationSupport.java:366)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.maybeInitializeAtBuildTime(ClassInitializationSupport.java:178)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.maybeInitializeAtBuildTime(ClassInitializationSupport.java:167)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.SVMHost.onTypeReachable(SVMHost.java:350)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisUniverse.onTypeReachable(AnalysisUniverse.java:676)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.lambda$new$0(AnalysisType.java:336)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        ... 17 more
Caused by: java.lang.NoClassDefFoundError: org/apache/log4j/Priority
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1161)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationSupport.ensureClassInitialized(ClassInitializationSupport.java:194)
        ... 26 more
Caused by: java.lang.ClassNotFoundException: org.apache.log4j.Priority
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:528)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageClassLoader.loadClass(NativeImageClassLoader.java:637)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:528)
        ... 29 more

```


### Steps to Reproduce

Create a new quarkus app:
quarkus create
Go into that project, add the following dependency (along with dependencyManagement block as in docs)
```
<dependency>
            <groupId>io.quarkiverse.googlecloudservices</groupId>
            <artifactId>quarkus-google-cloud-firebase-admin</artifactId>
 </dependency>
```

run native build command:
quarkus build --native --no-tests

--- a/docs/modules/ROOT/pages/includes/quarkus-google-cloud-storage.adoc
+++ b/docs/modules/ROOT/pages/includes/quarkus-google-cloud-storage.adoc
@@ -28,5 +28,215 @@ endif::add-copy-button-to-env-var[]
 |string
 |
 
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-use-grpc]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-use-grpc[`quarkus.google.cloud.storage.transport.use-grpc`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.transport.use-grpc+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Use the GRPC transport instead of the default HTTP transport.
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_USE_GRPC+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_USE_GRPC+++`
+endif::add-copy-button-to-env-var[]
+--
+|boolean
+|`false`
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-http-connect-timeout]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-http-connect-timeout[`quarkus.google.cloud.storage.transport.http-connect-timeout`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.transport.http-connect-timeout+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Storage HTTP transport connect timeout
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_HTTP_CONNECT_TIMEOUT+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_HTTP_CONNECT_TIMEOUT+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-http-read-timeout]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-http-read-timeout[`quarkus.google.cloud.storage.transport.http-read-timeout`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.transport.http-read-timeout+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Storage transport read timeout
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_HTTP_READ_TIMEOUT+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_HTTP_READ_TIMEOUT+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-initial-retry-delay-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-initial-retry-delay-millis[`quarkus.google.cloud.storage.retry.initial-retry-delay-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.initial-retry-delay-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Initial retry delay. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setInitialRetryDelay(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_INITIAL_RETRY_DELAY_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_INITIAL_RETRY_DELAY_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-retry-delay-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-retry-delay-millis[`quarkus.google.cloud.storage.retry.max-retry-delay-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.max-retry-delay-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Max retry delay. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setMaxRetryDelay(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_RETRY_DELAY_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_RETRY_DELAY_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-initial-rpc-timeout-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-initial-rpc-timeout-millis[`quarkus.google.cloud.storage.retry.initial-rpc-timeout-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.initial-rpc-timeout-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Initial RPC timeout. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setInitialRpcTimeout(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_INITIAL_RPC_TIMEOUT_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_INITIAL_RPC_TIMEOUT_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-rpc-timeout-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-rpc-timeout-millis[`quarkus.google.cloud.storage.retry.max-rpc-timeout-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.max-rpc-timeout-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Max RPC timeout. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setMaxRpcTimeout(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_RPC_TIMEOUT_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_RPC_TIMEOUT_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-logical-timeout-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-logical-timeout-millis[`quarkus.google.cloud.storage.retry.logical-timeout-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.logical-timeout-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Logical timeout. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setLogicalTimeout(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_LOGICAL_TIMEOUT_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_LOGICAL_TIMEOUT_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-total-timeout-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-total-timeout-millis[`quarkus.google.cloud.storage.retry.total-timeout-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.total-timeout-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Total timeout. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setTotalTimeout(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_TOTAL_TIMEOUT_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_TOTAL_TIMEOUT_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-attempts]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-attempts[`quarkus.google.cloud.storage.retry.max-attempts`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.max-attempts+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Maximum attempts. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setMaxAttempts(int)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_ATTEMPTS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_ATTEMPTS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
 |===
 

--- a/docs/modules/ROOT/pages/includes/quarkus-google-cloud-storage_quarkus.google.adoc
+++ b/docs/modules/ROOT/pages/includes/quarkus-google-cloud-storage_quarkus.google.adoc
@@ -28,5 +28,215 @@ endif::add-copy-button-to-env-var[]
 |string
 |
 
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-use-grpc]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-use-grpc[`quarkus.google.cloud.storage.transport.use-grpc`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.transport.use-grpc+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Use the GRPC transport instead of the default HTTP transport.
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_USE_GRPC+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_USE_GRPC+++`
+endif::add-copy-button-to-env-var[]
+--
+|boolean
+|`false`
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-http-connect-timeout]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-http-connect-timeout[`quarkus.google.cloud.storage.transport.http-connect-timeout`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.transport.http-connect-timeout+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Storage HTTP transport connect timeout
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_HTTP_CONNECT_TIMEOUT+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_HTTP_CONNECT_TIMEOUT+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-http-read-timeout]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-transport-http-read-timeout[`quarkus.google.cloud.storage.transport.http-read-timeout`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.transport.http-read-timeout+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Storage transport read timeout
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_HTTP_READ_TIMEOUT+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_TRANSPORT_HTTP_READ_TIMEOUT+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-initial-retry-delay-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-initial-retry-delay-millis[`quarkus.google.cloud.storage.retry.initial-retry-delay-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.initial-retry-delay-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Initial retry delay. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setInitialRetryDelay(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_INITIAL_RETRY_DELAY_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_INITIAL_RETRY_DELAY_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-retry-delay-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-retry-delay-millis[`quarkus.google.cloud.storage.retry.max-retry-delay-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.max-retry-delay-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Max retry delay. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setMaxRetryDelay(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_RETRY_DELAY_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_RETRY_DELAY_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-initial-rpc-timeout-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-initial-rpc-timeout-millis[`quarkus.google.cloud.storage.retry.initial-rpc-timeout-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.initial-rpc-timeout-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Initial RPC timeout. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setInitialRpcTimeout(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_INITIAL_RPC_TIMEOUT_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_INITIAL_RPC_TIMEOUT_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-rpc-timeout-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-rpc-timeout-millis[`quarkus.google.cloud.storage.retry.max-rpc-timeout-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.max-rpc-timeout-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Max RPC timeout. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setMaxRpcTimeout(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_RPC_TIMEOUT_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_RPC_TIMEOUT_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-logical-timeout-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-logical-timeout-millis[`quarkus.google.cloud.storage.retry.logical-timeout-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.logical-timeout-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Logical timeout. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setLogicalTimeout(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_LOGICAL_TIMEOUT_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_LOGICAL_TIMEOUT_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-total-timeout-millis]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-total-timeout-millis[`quarkus.google.cloud.storage.retry.total-timeout-millis`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.total-timeout-millis+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Total timeout. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setTotalTimeout(Duration)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_TOTAL_TIMEOUT_MILLIS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_TOTAL_TIMEOUT_MILLIS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
+a| [[quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-attempts]] [.property-path]##link:#quarkus-google-cloud-storage_quarkus-google-cloud-storage-retry-max-attempts[`quarkus.google.cloud.storage.retry.max-attempts`]##
+ifdef::add-copy-button-to-config-props[]
+config_property_copy_button:+++quarkus.google.cloud.storage.retry.max-attempts+++[]
+endif::add-copy-button-to-config-props[]
+
+
+[.description]
+--
+Maximum attempts. See `com.google.api.gax.retrying.RetrySettings.Builder++#++setMaxAttempts(int)`
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_ATTEMPTS+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++QUARKUS_GOOGLE_CLOUD_STORAGE_RETRY_MAX_ATTEMPTS+++`
+endif::add-copy-button-to-env-var[]
+--
+|int
+|
+
 |===
 

--- a/firebase-admin/runtime/pom.xml
+++ b/firebase-admin/runtime/pom.xml
@@ -25,8 +25,20 @@
             <groupId>com.google.firebase</groupId>
             <artifactId>firebase-admin</artifactId>
             <exclusions>
+                <exclusion>
+                    <groupId>commons-logging</groupId>
+                    <artifactId>commons-logging</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>javax.annotation</groupId>
+                    <artifactId>javax.annotation-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.checkerframework</groupId>
+                    <artifactId>checker-qual</artifactId>
+                </exclusion>
                 <!-- We exclude the grpc-netty-shaded library as grpc-netty is included via
-quarkus-google-cloud-common-grpc and Quarkus includes the netty library -->
+quarkus-google-cloud-common-grpc and Quarkus include the netty library -->
                 <exclusion>
                     <groupId>io.grpc</groupId>
                     <artifactId>grpc-netty-shaded</artifactId>

--- a/firebase-devservices/deployment/src/main/java/io/quarkiverse/googlecloudservices/firebase/deployment/testcontainers/FirebaseEmulatorContainer.java
+++ b/firebase-devservices/deployment/src/main/java/io/quarkiverse/googlecloudservices/firebase/deployment/testcontainers/FirebaseEmulatorContainer.java
@@ -20,7 +20,6 @@ import org.testcontainers.images.builder.ImageFromDockerfile;
 import org.testcontainers.images.builder.dockerfile.DockerfileBuilder;
 
 import com.github.dockerjava.api.DockerClient;
-
 import com.sun.security.auth.module.UnixSystem;
 
 /**

--- a/firebase-devservices/deployment/src/main/java/io/quarkiverse/googlecloudservices/firebase/deployment/testcontainers/FirebaseVersionResolver.java
+++ b/firebase-devservices/deployment/src/main/java/io/quarkiverse/googlecloudservices/firebase/deployment/testcontainers/FirebaseVersionResolver.java
@@ -1,13 +1,13 @@
 package io.quarkiverse.googlecloudservices.firebase.deployment.testcontainers;
 
-import com.google.api.client.json.gson.GsonFactory;
-import org.apache.commons.io.IOUtils;
-import org.apache.http.client.methods.HttpGet;
-import org.apache.http.impl.client.HttpClientBuilder;
-
 import java.io.IOException;
 import java.util.Map;
 import java.util.Objects;
+
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import com.google.api.client.json.gson.GsonFactory;
 
 /**
  * Resolves the version of the firebase tools based on the NPM registry in case version

--- a/firebase-devservices/deployment/src/test/java/io/quarkiverse/googlecloudservices/firebase/deployment/testcontainers/FirebaseEmulatorContainerCustomConfigTest.java
+++ b/firebase-devservices/deployment/src/test/java/io/quarkiverse/googlecloudservices/firebase/deployment/testcontainers/FirebaseEmulatorContainerCustomConfigTest.java
@@ -12,7 +12,6 @@ import java.net.http.HttpRequest;
 import java.net.http.HttpResponse;
 import java.util.Set;
 
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;

--- a/firebase-devservices/deployment/src/test/java/io/quarkiverse/googlecloudservices/firebase/deployment/testcontainers/FirebaseEmulatorContainerIntegrationTest.java
+++ b/firebase-devservices/deployment/src/test/java/io/quarkiverse/googlecloudservices/firebase/deployment/testcontainers/FirebaseEmulatorContainerIntegrationTest.java
@@ -21,7 +21,6 @@ import java.util.concurrent.TimeUnit;
 
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.testcontainers.junit.jupiter.Testcontainers;
 

--- a/storage/runtime/src/main/java/io/quarkiverse/googlecloudservices/storage/runtime/StorageConfiguration.java
+++ b/storage/runtime/src/main/java/io/quarkiverse/googlecloudservices/storage/runtime/StorageConfiguration.java
@@ -2,11 +2,12 @@ package io.quarkiverse.googlecloudservices.storage.runtime;
 
 import java.util.Optional;
 
+import org.threeten.bp.Duration;
+
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
 import io.smallrye.config.ConfigMapping;
 import io.smallrye.config.WithDefault;
-import org.threeten.bp.Duration;
 
 @ConfigMapping(prefix = "quarkus.google.cloud.storage")
 @ConfigRoot(phase = ConfigPhase.RUN_TIME)

--- a/storage/runtime/src/main/java/io/quarkiverse/googlecloudservices/storage/runtime/StorageProducer.java
+++ b/storage/runtime/src/main/java/io/quarkiverse/googlecloudservices/storage/runtime/StorageProducer.java
@@ -4,10 +4,6 @@ import java.io.IOException;
 import java.time.Duration;
 import java.util.Optional;
 
-import com.google.api.gax.retrying.RetrySettings;
-import com.google.cloud.TransportOptions;
-import com.google.cloud.grpc.GrpcTransportOptions;
-import com.google.cloud.http.HttpTransportOptions;
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.enterprise.inject.Default;
 import jakarta.enterprise.inject.Disposes;
@@ -15,7 +11,11 @@ import jakarta.enterprise.inject.Produces;
 import jakarta.inject.Inject;
 import jakarta.inject.Singleton;
 
+import com.google.api.gax.retrying.RetrySettings;
 import com.google.auth.Credentials;
+import com.google.cloud.TransportOptions;
+import com.google.cloud.grpc.GrpcTransportOptions;
+import com.google.cloud.http.HttpTransportOptions;
 import com.google.cloud.storage.Storage;
 import com.google.cloud.storage.StorageOptions;
 

